jupyter_scheduler/handlers.py

```diff
@@ -163,16 +164,16 @@ async def post(self):
     async def patch(self, job_id):
         payload = self.get_json_body()
 
-        if "status" not in payload:
-            raise tornado.web.HTTPError(500, "Field 'status' missing in request body")
+        status = payload.get("status", None)
+        status = Status(status) if status else None
+
+        if status and status != Status.STOPPED:
+            raise tornado.web.HTTPError(500, "Value of 'STOPPED' only allowed for field 'status'")
```

Comment:
1. I'd rename the status message to something like "Invalid value for field 'status'. Jobs can only be updated to status 'STOPPED' after creation."

2. Doesn't this technically allow you to stop a completed job? It looks like you're setting the status to 'STOPPED' as long as it's in the request, regardless of the current job's status. IMO, stopping a job only makes sense if the job is 'QUEUED' or 'IN PROGRESS'. If you do make this change, add a new corresponding error message as well, such as: "Invalid value for field 'status'. Only jobs that are either 'QUEUED' or 'IN PROGRESS' can be stopped."

Reply:
For 2), No, the scheduler API does not allow to stop a job.